pkg/tfshim/shim.go

```diff
@@ -213,14 +216,49 @@ type ProviderWithContext interface {
 	Provider
 
 	ConfigureWithContext(ctx context.Context, c ResourceConfig) error
-	DiffWithContext(ctx context.Context, t string, s InstanceState, c ResourceConfig) (InstanceDiff, error)
+
+	DiffWithContext(
+		ctx context.Context, t string, s InstanceState, c ResourceConfig, opts ...DiffOption,
+	) (InstanceDiff, error)
+
 	ApplyWithContext(ctx context.Context, t string, s InstanceState, d InstanceDiff) (InstanceState, error)
 	RefreshWithContext(ctx context.Context, t string, s InstanceState, c ResourceConfig) (InstanceState, error)
 
 	ReadDataDiffWithContext(ctx context.Context, t string, c ResourceConfig) (InstanceDiff, error)
 	ReadDataApplyWithContext(ctx context.Context, t string, d InstanceDiff) (InstanceState, error)
 }
 
+type DiffOptions struct {
+	IgnoreChanges IgnoreChanges
+}
+
+func NewDiffOptions(opts ...DiffOption) DiffOptions {
+	options := DiffOptions{}
+	for _, o := range opts {
+		o(&options)
+	}
+	return options
+}
```

Comment:
Nit: I think that this pattern is beneficial when you want composable, nested, and opaque settings. In this case, both sides of the contract are public: `DiffOption` and `DiffOptions`. I'm not sure what the benefit is here over just having the caller pass `DiffOptions` directly.

Reply:
Ah right, diffOptions shouldn't be public probably, I was in a bit of hurry.